### ISSUE TITLE
[v1.4] built-in adhoc host can't see other player

### ISSUE THREAD

**Issue by yanjingzhaisun** [hrydgard/ppsspp#9893]
Both players are using ppsspp 1.4.
Can ping each other.
Wifi wireless LAN network.

PC1 can't see the other player, but PC2 can. Server runs on PC1's ip address.
both player can connect to the server(can hear the lobby connection sound), and the outcome of PC2 can be seen in PC1's game(he can kill monsters, and harvest the items), it's just that PC1 can't see PC2's character.

if PC2 starts server, both players can't see each other.

I'm wondering if this is because that PC2 doesn't open a certain protocol or not?

the game: Toukidden.

**Comment by tywald** [comment]
Does it also happen on 1.4.2?
Double check that the MAC address in PPSSPP is different for each player.

**Comment by yanjingzhaisun** [comment]
MAC address is unique for each player.

Ways haven tried:

1. Updated to 1.4.2 stable (Win)(both), not working
2. Turn off Windows firewall(both), not working
3. PC2 sets Internet -> Network and Sharing Center -> Sharing Discover to be on, not working.
4. PC1 uninstalls Symantec, not working.

The weird thing is:
1. When PC1 host the sever, PC2 can see PC1(both character and status) in the lobby, but PC1 can’t see PC2 (neither character nor status) in the lobby. Though technically server knows that PC2 exist (because PC1 can only select 2 NPC instead of 3, which means one slot has been taken by other player)
2. When PC2 host the server, 2 players can’t see each other, neither character nor status.

This unsymmetrical outcome might means something, that’s why I wonder if PC2 has some ports / protocol not set correctly.

It’s LAN network. Router UPnp has been set on.

**Comment by JacobMrox** [comment]
You need to setup Upnp or DMZ on your router to allow connections, been an issue for a while because PPSSPP doesn't have a upnp feature.

**Comment by adenovan** [comment]
If its lan network there is no need for upnp. It seem some udp port is blocked on pc that cannot see other character , what game did you play?  Try using same port offset value, it strongly suspect the port is being use

**Comment by unknownbrackets** [comment]
Is this still an issue on the latest git build with recent networking improvements?

-[Unknown]

**Comment by unknownbrackets** [comment]
This issue is being closed because it hasn't been updated with feedback.

It's hard to tell when fixes in PPSSPP might fix other games, and sometimes certain settings or cheats may cause bugs that can't be fixed.

If you have more information or can confirm it still happens in the latest git builds, please reply to this issue and it'll be reopened.  If you have a new issue with the same game, just create a new issue instead.

-[Unknown]
